### PR TITLE
No more partial cmd in queue, less code

### PR DESCRIPTION
**Pull request by kisslorand**
### Requirements

BTT or MKS TFT

### Description

This PR eliminates partial or incomplete commands in the command queue and reduces code size.

### CODE REVIEW

File: TFT/src/User/API/interfaceCmd.c [hunk 1]
```diff
@@ -867,15 +861,8 @@ void sendQueueCmd(void)
...
-            uint8_t i = cmd_seen('T') ? cmd_value() : heatGetCurrentHotend();
-            if (cmd_seen('S'))
-            {
-              heatSyncTargetTemp(i, cmd_value());
-              heatSetSendWaiting(i, false);
-            }
-          }
-          break;
+            synchNoWaitHeating(cmd_seen('T') ? cmd_value() : heatGetCurrentHotend());
+           break;
```
Comment (digant73) [inline TFT/src/User/API/interfaceCmd.c:865]: one char less of indent

File: TFT/src/User/API/parseACK.c [hunk 2]
```diff
@@ -1078,7 +1078,9 @@ void parseACK(void)
         infoMachineSettings.autoReportTemp = ack_value();
 
         if (infoMachineSettings.autoReportTemp)
-          storeCmd("M155 ");
+        {
```
Comment (digant73) [inline TFT/src/User/API/parseACK.c:1081]: it should be better to not use {} in an "if" with only one instruction (as done in all the rest of the file)

**Comment by digant73** [comment]
According to your previous changes, the "else if" block at line 845 in interfaceCmd should be removed and so also the function updateCmd could be removed because no more used

**Comment by kisslorand** [comment]
> According to your previous changes, the "else if" block at line 845 in interfaceCmd (M155 case) should be removed and so also the function updateCmd could be removed because no more used

There are no such things left behind.

File: TFT/src/User/API/interfaceCmd.c (hunk 1 above)
Comment (kisslorand) [reply TFT/src/User/API/interfaceCmd.c:865]: Fixed.

File: TFT/src/User/API/parseACK.c (hunk 2 above)
Comment (kisslorand) [reply TFT/src/User/API/parseACK.c:1082]: Fixed.

**Comment by kisslorand** [comment]
@bigtreetech
Any reason this remains unmerged for so long?

**Comment by kisslorand** [comment]
Closing because it's also implemented in PR #2519.

**Pull request closed without merging**